qiskit/terra/pulse_schedules.ipynb

```diff
@@ -0,0 +1,845 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<img src=\"../../images/qiskit-heading.gif\" alt=\"Note: In order for images to show up in this jupyter notebook you need to select File => Trusted Notebook\" width=\"500 px\" align=\"left\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Pulse Schedules\n",
+    "\n",
+    "The `pulse` module allows quantum experiments to be described at the level of pulses. For IBMQ devices these are microwave pulses applied to our superconducting qubits. \n",
+    "\n",
+    "The pulse `Schedule`, `Instruction`, `Command` and `Channel` are the main objects of the pulse module. \n",
+    "\n",
+    "\n",
+    "\n",
+    "All instances of these objects are **immutable**, which results in:\n",
+    "- An API that is easier to reason about\n",
+    "- Objects that may be reused many times as components of parent objects without the problem of underlying children object's guarantees changing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 63,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from qiskit import pulse\n",
+    "from qiskit.pulse.channels import (DriveChannel, MeasureChannel, ControlChannel, \n",
+    "                                   AcquireChannel, MemorySlot, SnapshotChannel)\n",
+    "from qiskit.pulse.commands import (SamplePulse, FrameChange, PersistentValue,\n",
+    "                                   Acquire, Snapshot)\n",
+    "from qiskit.pulse import pulse_lib, Schedule"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Channels\n",
+    "\n",
+    "`Channel`s correspond to physical channels on the backend. `Channel`s are combined with `Command`s to form an `Instruction`.\n",
+    "\n",
+    "As all channels correspond to physical channels on a device, they have an `index` which specifies their corresponding device channel and a `buffer`, which is the buffering time between commands on the channel.\n",
+    "\n",
+    "The fundamental channel types are: \n",
+    "- `DriveChannel`: Qubit drive channel\n",
```

Comment:
I agree with this. Let's try to be a bit more descriptive since this is a new (and cool) feature. An alternative is to write a catalog of the key words used in Pulse in the header of this document.

Reply:
This is not documentation. This is a tutorial on how to use the API. This notebook is an analogy to the Quantum circuit notebook.

Reply:
I believe most people, in most cases, will find it difficult and/or frustrating to try to learn how to use something new if new concepts aren't given a definition, and confusing points aren't clarified. I hope that any tutorial or documentation can be written with the user in mind: how do we make their experience fun? How do we make Qiskit and any of its features feel straightforward and clear? Even if the concept is hard, can we make it feel easy? I know from experience it's a ton of work to do that! I think a compromise would be to leave this as is, and merge the PR only after that documentation is written, and then include some cross references.

Reply:
I'll work with both of you to have a tutorial showing off application-level usage of Pulse. Agree that best thing to do here is to merge as is and work on another intro-level version of a Pulse notebook.

Reply:
I'd prefer documentation got prioritized over another notebook, at least, that's what I would have found most helpful when I was trying to learn this. So, just a personal preference